poetry.lock

```diff
@@ -3567,17 +3558,17 @@ tests = ["coverage[toml] (==5.0.4)", "pytest (>=6.0.0,<7.0.0)"]
 
 [[package]]
 name = "pylint"
-version = "3.1.0"
+version = "3.2.0"
```

Comment:
I fear the problem is coming from here, but not sure. I'd try to pin Pylint to `3.1.0` in `pyproject.toml` and regenerate the lock.

If this is the actual reason, we could merge with the Pylint version pinned, but we should also consider fixing the errors.

Reply:
I applied the suggestion myself in a83e0c2, let's what happens

Reply:
Ok, hypothesis confirmed:
https://github.com/qiboteam/qibo/actions/runs/9099789800/job/25013165174#step:10:16
```py
src/qibo/backends/__init__.py:21:40: E0602: Undefined variable 'Backend' (undefined-variable)
src/qibo/backends/__init__.py:205:44: E0602: Undefined variable 'Backend' (undefined-variable)
```

Now the error is sensible, since related to this PR, and it should be fixed (it's just a missing import).

However, it could be worth to fix the previous ones as well.

Reply:
#1328